File: Data Structures & Algorithms/guess-number-higher-or-lower/submission-0.cpp
/** 
 * Forward declaration of guess API.
 * @param  num   your guess
 * @return 	     -1 if num is higher than the picked number
 *			      1 if num is lower than the picked number
 *               otherwise return 0
 * int guess(int num);
 */

class Solution {
public:
    int guessNumber(int n) {
        // Call the binary search helper function with the initial range [1, n]
        return binarySearch(1, n);
    }

    int binarySearch(long low, long high) {
        while (low <= high) {
            // Prevent potential overflow by using long or calculating mid this way
            long mid = low + (high - low) / 2;
            int res = guess(mid);
            
            if (res == 0) {
                return mid; // Found the picked number
            } else if (res == -1) {
                high = mid - 1; // The picked number is lower, reduce upper bound
            } else {
                low = mid + 1; // The picked number is higher, reduce lower bound
            }
        }
        return -1;
    }
};

File: Data Structures & Algorithms/ugly-number/submission-0.py
class Solution:
    def isUgly(self, n: int) -> bool:
        if n<1:
            return False
        if n==1:
            return True
        for i in range(2,n+1):
            if n%i!=0:
                continue
            else:
                primeCounter=0
                for j in range(1,i+1):
                    if i%j==0:
                        primeCounter+=1
                if primeCounter==2:
                    if i!=2 and i!=3 and i!=5:
                        return False
        return True

                    
        